clients/src/test/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventProcessorTest.java

```diff
@@ -177,6 +180,22 @@ public void testAssignmentChangeEventWithException() {
         assertInstanceOf(IllegalStateException.class, e.getCause());
     }
 
+    @Test
+    public void testResetOffsetEvent() {
+        Collection<TopicPartition> tp = Collections.singleton(new TopicPartition("topic", 0));
+        OffsetResetStrategy strategy = OffsetResetStrategy.LATEST;
+
+        ResetOffsetEvent event = mock(ResetOffsetEvent.class);
+        doReturn(strategy).when(event).offsetResetStrategy();
+        doReturn(tp).when(event).topicPartitions();
+        doReturn(RESET_OFFSET).when(event).type();
+        doReturn(CompletableFuture.completedFuture(null)).when(event).future();
+        setupProcessor(false);
+
+        processor.process(event);
+        verify(subscriptionState).requestOffsetReset(tp, strategy);
```

Comment:
How about rewrite the case like following, so we don't need to set `doReturn` for `event`? 

```java
@Test
    public void testResetOffsetEvent() {
        Collection<TopicPartition> tp = Collections.singleton(new TopicPartition("topic", 0));
        OffsetResetStrategy strategy = OffsetResetStrategy.LATEST;
        ResetOffsetEvent event = new ResetOffsetEvent(tp, strategy, 12345);

        setupProcessor(false);
        processor.process(event);
        verify(subscriptionState).requestOffsetReset(event.topicPartitions(), event.offsetResetStrategy());
    }
```

Reply:
Thanks! your approach is better.